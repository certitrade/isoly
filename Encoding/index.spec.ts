import * as isoly from "../index"

describe("Encoding", () => {
	const sample = "\n\r\t\b 0123456789ABCDEFGHIJKLMNOPQRSTUVXYZabcdefghijklmnopqrstuvwxyzÅÄÖåäöÜüß"
	it("is", () => {
		expect(isoly.Encoding.is("UTF-8")).toEqual(true)
		expect(isoly.Encoding.is("UTF-16")).toEqual(true)
		expect(isoly.Encoding.is("US-ASCII")).toEqual(false)
		expect(isoly.Encoding.is("ISO-8859-1")).toEqual(true)
		expect(isoly.Encoding.is("ISO 8859-1")).toEqual(false)
	})
	for (const encoding of ["UTF-8", "ISO-8859-1"] as ["UTF-8", "ISO-8859-1"])
		it(encoding, () => {
			const encoded = isoly.Encoding.encode(encoding, sample)
			expect(encoded).toMatchSnapshot()
			expect(isoly.Encoding.decode(encoding, encoded)).toEqual(sample)
		})
})
