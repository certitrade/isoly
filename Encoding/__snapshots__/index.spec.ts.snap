// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encoding ISO-8859-1 1`] = `
Uint8Array [
  10,
  13,
  9,
  8,
  32,
  48,
  49,
  50,
  51,
  52,
  53,
  54,
  55,
  56,
  57,
  65,
  66,
  67,
  68,
  69,
  70,
  71,
  72,
  73,
  74,
  75,
  76,
  77,
  78,
  79,
  80,
  81,
  82,
  83,
  84,
  85,
  86,
  88,
  89,
  90,
  97,
  98,
  99,
  100,
  101,
  102,
  103,
  104,
  105,
  106,
  107,
  108,
  109,
  110,
  111,
  112,
  113,
  114,
  115,
  116,
  117,
  118,
  119,
  120,
  121,
  122,
  197,
  196,
  214,
  229,
  228,
  246,
  220,
  252,
  223,
]
`;

exports[`Encoding UTF-8 1`] = `
Uint8Array [
  10,
  13,
  9,
  8,
  32,
  48,
  49,
  50,
  51,
  52,
  53,
  54,
  55,
  56,
  57,
  65,
  66,
  67,
  68,
  69,
  70,
  71,
  72,
  73,
  74,
  75,
  76,
  77,
  78,
  79,
  80,
  81,
  82,
  83,
  84,
  85,
  86,
  88,
  89,
  90,
  97,
  98,
  99,
  100,
  101,
  102,
  103,
  104,
  105,
  106,
  107,
  108,
  109,
  110,
  111,
  112,
  113,
  114,
  115,
  116,
  117,
  118,
  119,
  120,
  121,
  122,
  195,
  133,
  195,
  132,
  195,
  150,
  195,
  165,
  195,
  164,
  195,
  182,
  195,
  156,
  195,
  188,
  195,
  159,
]
`;
